web-app/src/app/screens/Feed.tsx

```diff
@@ -0,0 +1,561 @@
+import * as React from 'react';
+import { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import {
+  Box,
+  Container,
+  CssBaseline,
+  Typography,
+  Button,
+  Grid,
+  Chip,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableRow,
+  colors,
+} from '@mui/material';
+import {
+  ContentCopy,
+  ContentCopyOutlined,
+  Download,
+  DownloadOutlined,
+  ErrorOutlineOutlined,
+  LaunchOutlined,
+  OpenInNewOutlined,
+  ReportOutlined,
+  ReportProblemOutlined,
+  WarningAmberOutlined,
+} from '@mui/icons-material';
+import '../styles/SignUp.css';
+import '../styles/FAQ.css';
+import { ContentBox } from '../components/ContentBox';
+import { useAppDispatch } from '../hooks';
+import { loadingFeed } from '../store/feed-reducer';
+import { selectUserProfile } from '../store/profile-selectors';
+import {
+  selectFeedData,
+  selectFeedLoadingStatus,
+  selectGTFSFeedData,
+  selectGTFSRTFeedData,
+} from '../store/feed-selectors';
+import { loadingDataset } from '../store/dataset-reducer';
+import {
+  selectBoundingBoxFromLatestDataset,
+  selectDatasetsData,
+  selectLatestDatasetsData,
+} from '../store/dataset-selectors';
+import { Map } from '../components/Map';
+
+export default function Feed(): React.ReactElement {
+  const { feedId } = useParams();
+  const user = useSelector(selectUserProfile);
+  const feedLoadingStatus = useSelector(selectFeedLoadingStatus);
+  const feedType = useSelector(selectFeedData)?.data_type;
+  const feed =
+    feedType === 'gtfs'
+      ? useSelector(selectGTFSFeedData)
+      : useSelector(selectGTFSRTFeedData);
+  const datasets = useSelector(selectDatasetsData);
+  const latestDataset = useSelector(selectLatestDatasetsData);
+  const boundingBox = useSelector(selectBoundingBoxFromLatestDataset);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    if (user?.accessToken !== undefined && feedId !== undefined) {
+      dispatch(loadingFeed({ feedId, accessToken: user?.accessToken }));
+      dispatch(loadingDataset({ feedId, accessToken: user?.accessToken }));
+    }
+  }, []);
+
+  const renderFeedSummary = (): JSX.Element => (
```

Comment:
I love the methods made to render this content, it makes the code much more structured. I'd like to see these methods in their own component to bring this file from ~600 lines to ~250 lines

Reply:
Pushed this change.